samples/DevTestLabs/SimpleUI/ClientApp/src/App.jsx

```diff
@@ -0,0 +1,132 @@
+            loading: true,
+            loggedInUser: null,
+        };
+
+        // Intialize the MSAL application object
+        this.publicClientApplication = new msal.PublicClientApplication(this.msalConfig);
+    }
+
+    async componentDidMount() {
+        await this.populateVirtualMachines();
+    }
+
+    render() {
+        let contents = this.state.loading
+            ? <Spinner label="Loading virtual machines..." />
+            : this.renderVirtualMachinesTable(this.state.virtualMachines);
+
+        const theme = getTheme();
+
+        return (
+            <div style={{ boxShadow: theme.effects.elevation8 }}>
+                <div className="header" style={{ padding: "10px" }}>
+                    <Text variant="xxLarge">Lab Virtual Machines</Text>
+                </div>
+                {contents}
+            </div>
+        );
+    }
+
+    renderVirtualMachinesTable(virtualMachines) {
+        return (
+            <DetailsList
+                items={virtualMachines.map(vm => {
+                    return {
+                        "Name": vm.name,
+                        "Owner": vm.ownerUserPrincipalName,
+                        "Location": vm.location,
+                        "Change Ownership": this.renderButton(vm)
+                    };
+                })}
+                checkboxVisibility={CheckboxVisibility.hidden}
+            />
+        );
+    }
+
+    renderButton(virtualMachine) {
+        return <VirtualMachineRow
+            vmOwner={virtualMachine.ownerUserPrincipalName}
+            loggedInUser={this.state.loggedInUser}
+            unclaim={() => this.unclaimVirtualMachine(virtualMachine.name)}
+            claim={() => this.claimVirtualMachine(virtualMachine.name)}
+        />;
+    }
+
+    async claimVirtualMachine(vmName) {
+        const token = await this.getAcccessToken();
+        await fetch(`virtualmachines/claim/${vmName}`,
+            {
+                method: "POST",
+                headers: {
+                    "Authorization": `Bearer ${token.accessToken}`
+                }
+            });
+        await this.populateVirtualMachines();
+    }
+
+    async unclaimVirtualMachine(vmName) {
+        const token = await this.getAcccessToken();
+        await fetch(`virtualmachines/unclaim/${vmName}`,
+            {
+                method: "POST",
+                headers: {
+                    "Authorization": `Bearer ${token.accessToken}`
+                }
+            });
+        await this.populateVirtualMachines();
+    }
+
+    async populateVirtualMachines() {
+        const token = await this.getAcccessToken();
+        const response = await fetch("virtualmachines",
+            {
+                headers: {
+                    "Authorization": `Bearer ${token.accessToken}`
+                }
+            });
+        const data = await response.json();
+        this.setState({ virtualMachines: data, loading: false });
+    }
+
+    async getAcccessToken() {
```

Comment:
Rename this method, or update it to only return the access token itself (rather than an object containing the access token)